source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/connect/ServerTypeSelectControl.java

```diff
@@ -245,4 +382,42 @@ public URI getServerURI() {
             return serverURI;
         }
     }
+
+    private class PropsalProvider implements IContentProposalProvider {
+        private String[] proposals = new String[0];
+
+        public void setProposals(final String[] proposals) {
+            if (proposals == null) {
+                this.proposals = new String[0];
+            } else {
+                this.proposals = proposals;
+            }
+        }
+
+        @Override
+        public IContentProposal[] getProposals(final String contents, final int position) {
+            // Work-around to make sure that the popup doesn't open when the
+            // user has opened the dropdown of the combo. By returning an empty
+            // list, the popup won't be shown.
+            if (ignoreServerChangeEvents || serverCombo.getListVisible()) {
+                return new IContentProposal[0];
+            }
+
+            final List<IContentProposal> list = new ArrayList<IContentProposal>(proposals.length);
+            if (contents == null || contents.length() == 0) {
+                for (int i = 0; i < proposals.length; i++) {
+                    list.add(new ContentProposal(proposals[i]));

```

Comment:
This method is not called by us. It is called by the SWT control adapter. Therefore, I prefer to check for null. I can use StringHelpers here.